### DISCUSSION TITLE
defining-custom-logging-messages-with-loggermessage-define-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#76] [Announcements]
# defining-custom-logging-messages-with-loggermessage-define-in-asp-net-core/

In this post I take a look at one of the helper methods in the ASP.NET Core logging library, and how you can use it to efficiently log messages in your app

https://andrewlock.net/defining-custom-logging-messages-with-loggermessage-define-in-asp-net-core/

**Comment by driekus77** [comment]
Nice article!

I've tested the difference in performance using BenchmarkDotNet:
// * Summary *

BenchmarkDotNet=v0.13.5, OS=Windows 11 (10.0.22621.1265/22H2/2022Update/SunValley2)
Intel Core i7-8850H CPU 2.60GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
.NET SDK=7.0.201
  [Host]     : .NET 7.0.3 (7.0.323.6910), X64 RyuJIT AVX2
  Job-YDVQGO : .NET 7.0.3 (7.0.323.6910), X64 RyuJIT AVX2

Platform=X64  Server=True  

|              Method |     Mean |   Error |  StdDev | Ratio | RatioSD |
|-------------------- |---------:|--------:|--------:|------:|--------:|
|    ILogger_LogDebug | 195.6 ns | 2.60 ns | 2.43 ns |  1.00 |    0.00 |
| LoggerMessage_Debug | 122.1 ns | 2.39 ns | 3.87 ns |  0.63 |    0.02 |

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 13 2017, 07:53, in reply to [nemke](https://disqus.com/by/nemke/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/76#discussioncomment-6024992)</em>

---

<p>Thanks!</p>

[a comment by an automated account is not shown]
